File: python/run_server.py
from ezpysocket import ezpysocket as ps
import cv2
import numpy as np

if __name__ == "__main__":

    # Init
    s = ps.EzPySocket(debug=True)

    # Load image
    img = cv2.imread("lena.jpg", 1)

# Receive examples
    print("Receiving data...")

    print("Bool 1 : ", s.receive_bool())
    print("Bool 2 : ", s.receive_bool())
    print("Bool 3 : ", s.receive_bool())
    print("Bool 4 : ", s.receive_bool())

    data = s.receive_string()
    print("String : ", data)

    data = s.receive_int()
    print("Int : ", data)

    data = s.receive_float()
    print("Float : ", data)

    data = s.receive_int_list()
    print("Int List : ", data)

    data = s.receive_float_list()
    print("Float List : ", data)

    img = s.receive_image()
    print("Showing image...")
    cv2.imshow("Python Server Received image", img)
    cv2.waitKey(0)
    print("Press a key to start sending from Python server to Python client...")


# Send examples
    print("Sending data...")

    s.send_bool(True)
    s.send_bool(False)
    s.send_bool(0)
    s.send_bool(1)

    s.send_string("It was the only way.")

    s.send_int(1024)
    s.send_float(3.14)

    # Send an int list
    a = []
    for i in range(100):
        a.append(i)
    s.send_int_list(a)

    # Send a float list
    a = np.arange(0, 10, 0.25)
    b = [i for i in a]
    s.send_float_list(b)

    s.send_image(img)

    s.disconnect()

File: cpp/examples/2.Tokens/run_server.cpp
#include "ezcppsocket.h"

int main(int argc, char const *argv[])
{
	EzCppSocket c = EzCppSocket("127.0.0.1", 10000, 2, 1, false, true, 1, true, 2000000U, std::pair<std::string, std::string>("start", "end"));

	// Receiving
	std::cout << "Receiving data...\n";

	auto res = c.readBool();
	std::cout << "Bool 1 : " << res.first << "," << res.second << "\n";

	res = c.readBool();
	std::cout << "Bool 2 : " << res.first << "," << res.second << "\n";

	res = c.readBool();
	std::cout << "Bool 3 : " << res.first << "," << res.second << "\n";

	res = c.readBool();
	std::cout << "Bool 4 : " << res.first << "," << res.second << "\n";

	std::cout << "String : " << c.readString() << "\n";

	std::cout << "Int : " << c.readInt() << "\n";
	std::cout << "Float : " << c.readFloat() << "\n";

	std::cout << "Int List : ";
	std::vector<int> int_list = c.readIntList();
	for (int &list_item : int_list)
	{
		std::cout << list_item << " ";
	}
	std::cout << "\n";

	std::cout << "Float List : ";
	std::vector<float> float_list = c.readFloatList();
	for (auto list_item : float_list)
	{
		std::cout << list_item << " ";
	}
	std::cout << "\n";

	cv::Mat recv_img = c.readImage();
	cv::imshow("Cpp Server Received Image", recv_img);
	std::cout << "Press a key to start sending data from server ...\n";
	cv::waitKey(0);

	// Sending
	std::cout << "Sending data...\n";

	c.sendBool(true);
	c.sendBool(false);
	c.sendBool(0);
	c.sendBool("wtv");

	c.sendString("Hello there, Dr. Strange");

	c.sendInt(512);
	c.sendFloat(3.14);

	std::vector<int> vi(7, 7);
	c.sendIntList(vi);

	std::vector<float> vf(10, 3.14);
	c.sendFloatList(vf);

	cv::Mat img = cv::imread("lena.jpg");
	c.sendImage(img);

	c.Disconnect();

	return 0;
}